src/risclog/sqlalchemy/model.py

```diff
@@ -51,8 +51,12 @@ def delete(self):
         zope.component.getUtility(IDatabase).delete(self)
 
     @classmethod
-    def query(cls):
-        return zope.component.getUtility(IDatabase).query(cls)
+    def query(cls, *args):
+        db = zope.component.getUtility(IDatabase)
+        if args:
+            selects = list(getattr(cls, a) for a in args)
+            return db.query(*selects).select_from(cls)
```

Comment:
This is a feature, which is not mentioned in the changes and also not tested. Is this feature intended to be part of this PR? From reading the SQLAlchemy documentation it seems, that the selecting is [changed a bit](https://docs.sqlalchemy.org/en/14/changelog/migration_20.html#select-no-longer-accepts-varied-constructor-arguments-columns-are-passed-positionally). This makes it even more important to test code around that feature.

Reply:
This feature also needs documentation, from reading the code it is not immediately clear what it does.

Reply:
Its ok for me to take that out and I'll reimplement it in the project I need it for. This is memory optimization, we load millions of objects into memory in a project and it makes a difference whether I load the whole object or just the attributes I need. But the decision is made at a higher level, so the API (`<model>.query(...)`) has to remain the same and I can't just do a simple `select` that easily.

Reply:
If it is only used in a single project, I'd prefer to implement it over there.